cupy/core/_routines_math.pyx

```diff
@@ -0,0 +1,531 @@
+     'LL->L', 'qq->q', 'QQ->Q',
+     ('ee->e', 'out0 = powf(in0, in1)'),
+     ('ff->f', 'out0 = powf(in0, in1)'),
+     ('dd->d', 'out0 = pow(in0, in1)'),
+     ('FF->F', 'out0 = pow(in0, in1)'),
+     ('DD->D', 'out0 = pow(in0, in1)')),
+    'out0 = rint(pow((double)in0, (double)in1))',
+    doc='''Computes ``x1 ** x2`` elementwise.
+
+    .. seealso:: :data:`numpy.power`
+
+    ''')
+
+
+_subtract = create_arithmetic(
+    'subtract', '-', '^',
+    '''Subtracts arguments elementwise.
+
+    .. seealso:: :data:`numpy.subtract`
+
+    ''')
+
+
+_true_divide = create_ufunc(
+    'cupy_true_divide',
+    ('bb->d', 'BB->d', 'hh->d', 'HH->d', 'ii->d', 'II->d', 'll->d', 'LL->d',
+     'qq->d', 'QQ->d', 'ee->e', 'ff->f', 'dd->d', 'FF->F', 'DD->D'),
+    'out0 = (out0_type)in0 / (out0_type)in1',
+    doc='''Elementwise true division (i.e. division as floating values).
+
+    .. seealso:: :data:`numpy.true_divide`
+
+    ''')
+
+
+if six.PY3:
+    _divide = _true_divide
+
+
+_floor_divide = create_ufunc(
+    'cupy_floor_divide',
+    ('bb->b', 'BB->B', 'hh->h', 'HH->H', 'ii->i', 'II->I', 'll->l', 'LL->L',
+     'qq->q', 'QQ->Q', 'ee->e', 'ff->f', 'dd->d'),
+    'out0 = _floor_divide(in0, in1)',
+    doc='''Elementwise floor division (i.e. integer quotient).
+
+    .. seealso:: :data:`numpy.floor_divide`
+
+    ''')
+
+
+_remainder = create_ufunc(
+    'cupy_remainder',
+    ('bb->b', 'BB->B', 'hh->h', 'HH->H', 'ii->i', 'II->I', 'll->l', 'LL->L',
+     'qq->q', 'QQ->Q',
+     ('ee->e', 'out0 = in0 - _floor_divide(in0, in1) * in1'),
+     ('ff->f', 'out0 = in0 - _floor_divide(in0, in1) * in1'),
+     ('dd->d', 'out0 = in0 - _floor_divide(in0, in1) * in1')),
+    'out0 = (in0 - _floor_divide(in0, in1) * in1) * (in1 != 0)',
+    doc='''Computes the remainder of Python division elementwise.
+
+    .. seealso:: :data:`numpy.remainder`
+
+    ''')
+
+
+_absolute = create_ufunc(
+    'cupy_absolute',
+    (('?->?', 'out0 = in0'),
+     'b->b', ('B->B', 'out0 = in0'), 'h->h', ('H->H', 'out0 = in0'),
+     'i->i', ('I->I', 'out0 = in0'), 'l->l', ('L->L', 'out0 = in0'),
+     'q->q', ('Q->Q', 'out0 = in0'),
+     ('e->e', 'out0 = fabsf(in0)'),
+     ('f->f', 'out0 = fabsf(in0)'),
+     ('d->d', 'out0 = fabs(in0)'),
+     ('F->f', 'out0 = abs(in0)'),
+     ('D->d', 'out0 = abs(in0)')),
+    'out0 = in0 > 0 ? in0 : -in0',
+    doc='''Elementwise absolute value function.
+
+    .. seealso:: :data:`numpy.absolute`
+
+    ''')
+
+
+_sqrt = create_ufunc(
+    'cupy_sqrt',
+    ('e->e', 'f->f', 'd->d', 'F->F', 'D->D'),
+    'out0 = sqrt(in0)',
+    doc='''Elementwise square root function.
+
+    .. seealso:: :data:`numpy.sqrt`
+
+    ''')
+
+
+_clip = create_ufunc(
+    'cupy_clip',
+    ('???->?', 'bbb->b', 'BBB->B', 'hhh->h', 'HHH->H', 'iii->i', 'III->I',
+     'lll->l', 'LLL->L', 'qqq->q', 'QQQ->Q', 'eee->e', 'fff->f', 'ddd->d'),
+    'out0 = in0 < in1 ? in1 : (in0 > in2 ? in2 : in0)')
+
+
+# Variables to expose to Python
```

Comment:
As it's seemingly not possible to declare a variable as `cpdef`, I declare those for Cython and those for Python with different names (former with `_`).